### PR TITLE
Adds liking videos and mostly functional profile page

### PR DESCRIPTION
**Pull request by WJehee**
Liking videos works although there are still some improvements to be made regarding the verification of the trustchain blocks.
The UI is functional, but we still need to make everything update on the fly (not only when the app is restarted).

### CODE REVIEW

File: app/build.gradle
Comment (OrestisKan) [inline app/build.gradle:45]: This looks like a testing thing, shouldn't be merged

File: detoks/src/main/java/nl/tudelft/trustchain/detoks/DeToksCommunity.kt
Comment (OrestisKan) [inline detoks/src/main/java/nl/tudelft/trustchain/detoks/DeToksCommunity.kt:11]: Looks like a redundant import since ...attestation.trustachain.* is already imported

File: detoks/src/main/java/nl/tudelft/trustchain/detoks/DeToksCommunity.kt
Comment (OrestisKan) [inline detoks/src/main/java/nl/tudelft/trustchain/detoks/DeToksCommunity.kt:129]: Not sure why an empty date constructor is needed here, since itn not the current datetime it could be added later when used

File: detoks/src/main/java/nl/tudelft/trustchain/detoks/DeToksCommunity.kt
Comment (OrestisKan) [inline detoks/src/main/java/nl/tudelft/trustchain/detoks/DeToksCommunity.kt:143]: Is this fixed? Also don't submit commented out code (talking about the next block of commented out code)

File: detoks/src/main/java/nl/tudelft/trustchain/detoks/DeToksCommunity.kt
Comment (OrestisKan) [inline detoks/src/main/java/nl/tudelft/trustchain/detoks/DeToksCommunity.kt:165]: You can use 
 
return liker in getLikes(...)

Or 

return getLikes(...).contains(liker)

This doesn't seem like the most idiomatic approach, and I'm pretty sure it should be slower and also harder to read

File: detoks/src/main/java/nl/tudelft/trustchain/detoks/DeToksMainActivity.kt
Comment (OrestisKan) [inline detoks/src/main/java/nl/tudelft/trustchain/detoks/DeToksMainActivity.kt:17]: Looks like a redundant import since its covered by the above

File: detoks/src/main/java/nl/tudelft/trustchain/detoks/DeToksMainActivity.kt
Comment (OrestisKan) [inline detoks/src/main/java/nl/tudelft/trustchain/detoks/DeToksMainActivity.kt:81]: Don't commit commented out code

File: detoks/src/main/java/nl/tudelft/trustchain/detoks/TorrentEngine.kt
Comment (OrestisKan) [inline detoks/src/main/java/nl/tudelft/trustchain/detoks/TorrentEngine.kt:70]: Don't commit commented out code

File: detoks/src/main/java/nl/tudelft/trustchain/detoks/TorrentManager.kt
Comment (OrestisKan) [inline detoks/src/main/java/nl/tudelft/trustchain/detoks/TorrentManager.kt:47]: Delete

File: detoks/src/main/java/nl/tudelft/trustchain/detoks/TorrentManager.kt
Comment (OrestisKan) [inline detoks/src/main/java/nl/tudelft/trustchain/detoks/TorrentManager.kt:154]: Delete

File: detoks/src/main/java/nl/tudelft/trustchain/detoks/TorrentManager.kt
Comment (OrestisKan) [inline detoks/src/main/java/nl/tudelft/trustchain/detoks/TorrentManager.kt:450]: Don't commit commented out code

File: detoks/src/main/java/nl/tudelft/trustchain/detoks/fragments/LikedListFragment.kt
Comment (OrestisKan) [inline detoks/src/main/java/nl/tudelft/trustchain/detoks/fragments/LikedListFragment.kt:19]: Is this done?

File: detoks/src/main/java/nl/tudelft/trustchain/detoks/fragments/ProfileFragment.kt
Comment (OrestisKan) [inline detoks/src/main/java/nl/tudelft/trustchain/detoks/fragments/ProfileFragment.kt:53]: Lools like the todos are not resolved

File: detoks/src/main/java/nl/tudelft/trustchain/detoks/fragments/VideosListFragment.kt
Comment (OrestisKan) [inline detoks/src/main/java/nl/tudelft/trustchain/detoks/fragments/VideosListFragment.kt:19]: Unfinished?

File: detoks/src/main/java/nl/tudelft/trustchain/detoks/DeToksCommunity.kt
Comment (WJehee) [reply detoks/src/main/java/nl/tudelft/trustchain/detoks/DeToksCommunity.kt:11]: It seems like the import is required, although im not sure why, removed the other redudant imports though

**Comment by WJehee** [comment]
@OrestisKan 
Please excuse the sloppiness with the TODO's and commented code.
All of the issues should be fixed now, if there is anything else, please let us know!

**Comment by synctext** [comment]
tested an isolated .APK with these new features, looks OK and is stable for me.

File: detoks/src/main/java/nl/tudelft/trustchain/detoks/DeToksCommunity.kt
Comment (OrestisKan) [inline detoks/src/main/java/nl/tudelft/trustchain/detoks/DeToksCommunity.kt:160]: Is this done ?

File: detoks/src/main/java/nl/tudelft/trustchain/detoks/TorrentManager.kt
Comment (OrestisKan) [inline detoks/src/main/java/nl/tudelft/trustchain/detoks/TorrentManager.kt:157]: AA ?

File: detoks/src/main/java/nl/tudelft/trustchain/detoks/TorrentManager.kt
Comment (OrestisKan) [inline detoks/src/main/java/nl/tudelft/trustchain/detoks/TorrentManager.kt:205]: is this french for me ?

File: detoks/src/main/java/nl/tudelft/trustchain/detoks/TorrentManager.kt
Comment (OrestisKan) [inline detoks/src/main/java/nl/tudelft/trustchain/detoks/TorrentManager.kt:202]: is this french for me ?

File: detoks/src/main/java/nl/tudelft/trustchain/detoks/TorrentManager.kt
Comment (OrestisKan) [inline detoks/src/main/java/nl/tudelft/trustchain/detoks/TorrentManager.kt:211]: delete commented out code

File: detoks/src/main/java/nl/tudelft/trustchain/detoks/TorrentManager.kt
Comment (OrestisKan) [inline detoks/src/main/java/nl/tudelft/trustchain/detoks/TorrentManager.kt:220]: delete commented out code

File: detoks/src/main/java/nl/tudelft/trustchain/detoks/TorrentManager.kt
Comment (OrestisKan) [inline detoks/src/main/java/nl/tudelft/trustchain/detoks/TorrentManager.kt:279]: can be simplified with inline if  ? :

**Review by OrestisKan — COMMENTED** [review]
No big problems were found, the main point of improvement would be the logging messages since a lot of times its not clear or its just an output of a function thus it will be hard to debug since we'll have to go back to the code to figure out which line generated what based on previous messages. I will create an apk soon and test it and then approve if its alright

**Comment by OrestisKan** [comment]
> No big problems were found, the main point of improvement would be the logging messages since a lot of times its not clear or its just an output of a function thus it will be hard to debug since we'll have to go back to the code to figure out which line generated what based on previous messages. I will create an apk soon and test it and then approve if its alright

Created the .apk but for some reason google play protect blocks it even throw I allow installation of unknown apps from my file, looking into it now

**Comment by WJehee** [comment]
@OrestisKan all the things mentioned should be fixed at this point

File: detoks/src/main/java/nl/tudelft/trustchain/detoks/DeToksMainActivity.kt
Comment (OrestisKan) [reply detoks/src/main/java/nl/tudelft/trustchain/detoks/DeToksMainActivity.kt:69]: this is not resolved

File: detoks/src/main/java/nl/tudelft/trustchain/detoks/fragments/ProfileFragment.kt
Comment (OrestisKan) [inline detoks/src/main/java/nl/tudelft/trustchain/detoks/fragments/ProfileFragment.kt:29]: not needed

**Review by OrestisKan — APPROVED** [review]
Seems to be working, also no major coding improvements required.
An improvement would be, as you can see in the attached picture to not just show the magnet link in the liked video but something more miningful, but this is not a major issue anw.

image

Also it seems to me that everything is indeed updated on the fly now and not when the app is restarted. so its alright I guess.

**Comment by WJehee** [comment]
@OrestisKan I have made an issue for your request regarding the magnet link, we will look into it.
Is there anything else (besides the last few open discussions) left to do before we can merge?

**Comment by OrestisKan** [comment]
Request my review when you're done.

**Review by OrestisKan — COMMENTED** [review]
Magnet link issue is still there !

**Comment by WJehee** [comment]
Fixed! @OrestisKan

**Review by OrestisKan — APPROVED** [review]
Much better. I guess that numbers is the video name which should be sufficient. Approved!

WhatsApp Image 2023-04-10 at 18 21 30

**Comment by WJehee** [comment]
Indeed, these are the video names!
When will this be merged? because we don't have permission to do so

**Comment by OrestisKan** [comment]
> Indeed, these are the video names! When will this be merged? because we don't have permission to do so

@WJehee Since I merged another PR  now please fix the merge conflicts and request a new review, then I'll merge it.

**Pull request closed without merging**
